tests/unit/tf/transformers/test_block.py

```diff
@@ -25,6 +25,59 @@ def test_import():
     assert transformers is not None
 
 
+@pytest.mark.parametrize("run_eagerly", [True])
+def test_retrieval_transformer(sequence_testing_data: Dataset, run_eagerly):
+
+    seq_schema = sequence_testing_data.schema.select_by_tag(Tags.SEQUENCE).select_by_tag(
+        Tags.CATEGORICAL
+    )
+
+    target = sequence_testing_data.schema.select_by_tag(Tags.ITEM_ID).column_names[0]
+    predict_next = mm.SequencePredictNext(schema=seq_schema, target=target)
+    loader = Loader(sequence_testing_data, batch_size=8, shuffle=False, transform=predict_next)
+
+    query_schema = seq_schema
+    output_schema = seq_schema.select_by_name(target)
+
+    query_encoder = mm.Encoder(
+        mm.InputBlockV2(
+            query_schema,
+            embeddings=mm.Embeddings(
+                query_schema.select_by_tag(Tags.CATEGORICAL), sequence_combiner=None
+            ),
+        ),
+        GPT2Block(d_model=48, n_head=4, n_layer=2, pre=mm.ReplaceMaskedEmbeddings()),
```

Comment:
The `ReplaceMaskedEmbeddings()` is only needed when `SequenceMaskRandom()` is used (Masked Language Modeling. As we are using `SequencePredictNext` here for Causal Language Modeling, there will be no masked items to have embeddings replaced

Reply:
Thanks @gabrielspmoreira that makes sense. I was copying this from another test that used GPT2Block and didn't make the connection with the `SequenceMaskRandom`.

I guess including it in a case like this where there are no masks doesn't affet the model? (in other words does it act as a no-op). Or would it be worth adding something to `ReplaceMaskedEmbeddings` to raise an exception if no masks are found?